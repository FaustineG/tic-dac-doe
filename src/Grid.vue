<template>
  <div class="grid">
    <template v-for="(n, i) of gameArray.length" :key="i">
      <template v-for="(m, j) of gameArray.length" :key="i + ',' + j">
        <input
          type="text"
          class="cell"
          data-cy="cell"
          readonly
          @click="() => onClick(i, j)"
          :value="gameArray[i][j]"
        />
      </template>
    </template>
  </div>
</template>

<script setup lang="ts">
defineProps<{ gameArray: string[][] }>()
const emits = defineEmits<{
  click: [i: number, j: number]
}>()

const onClick = (i: number, j: number) => emits('click', i, j)

</script>
<style lang="scss" scoped>
.grid {
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  gap: 1em;
  width: fit-content;
  .cell {
    border: none;
    width: 160px;
    height: 160px;
    display: flex;
    align-items: center;
    justify-content: center;
    font-size: 80px;
    text-align: center;
    cursor: pointer;
    border-radius: 5%;
  }
  .cell:active {
    outline: none;
  }
}

</style>
