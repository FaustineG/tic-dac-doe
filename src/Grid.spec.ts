import { describe, expect, it } from 'vitest'
import { mount } from '@vue/test-utils'
import Grid from './Grid.vue'

describe('Grid component', () => {
  it('should match snapshot', () => {
    const wrapper = mount(Grid, {
      props: {
        gameArray: [
          ['X', ''],
          ['', 'O'],
        ],
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  
  it('should display the game array', () => {
    const wrapper = mount(Grid, {
      props: {
        gameArray: [
          ['X', ''],
          ['', 'O'],
        ],
      },
    })
    expect(
      wrapper.findAll('[data-cy="cell"]').map((e) => (e.element as HTMLInputElement).value),
    ).toStrictEqual(['X', '', '', 'O'])
  })

  it('should emit event on click', async () => {
    const wrapper = mount(Grid, {
      props: {
        gameArray: [
          ['', ''],
          ['', ''],
        ],
      },
    })
    expect(
      wrapper.findAll('[data-cy="cell"]').map((e) => (e.element as HTMLInputElement).value),
    ).toStrictEqual(['', '', '', ''])
    await wrapper.findAll('[data-cy="cell"]')[0].trigger('click')
    expect(wrapper.emitted()['click']).toHaveLength(1)
  })
})
