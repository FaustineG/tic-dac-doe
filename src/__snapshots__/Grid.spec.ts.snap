// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Grid component > should match snapshot 1`] = `"<div data-v-bc14fce6="" class="grid"><input data-v-bc14fce6="" type="text" class="cell" data-cy="cell" readonly="" value="X"><input data-v-bc14fce6="" type="text" class="cell" data-cy="cell" readonly="" value=""><input data-v-bc14fce6="" type="text" class="cell" data-cy="cell" readonly="" value=""><input data-v-bc14fce6="" type="text" class="cell" data-cy="cell" readonly="" value="O"></div>"`;
